src/main/java/org/jabref/logic/citation/service/SearchCitationsRelationsService.java

```diff
@@ -0,0 +1,60 @@
+package org.jabref.logic.citation.service;
+
+import java.util.List;
+
+import org.jabref.logic.citation.repository.BibEntryRelationsRepository;
+import org.jabref.logic.importer.fetcher.CitationFetcher;
+import org.jabref.model.entry.BibEntry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SearchCitationsRelationsService {
```

Comment:
`Fetching` and `Repository` logic can now be injected in orchestration logic that should not vary in next PR.

This should also make it possible to configure a citation search service based on the execution context. This approach can also enable new features like offering the user the possibility to choose between multiple fetchers targeting another online search engine.